src/librbd/operation/SnapshotCreateRequest.cc

```diff
@@ -294,8 +294,9 @@ void SnapshotCreateRequest<I>::update_snap_context() {
          image_ctx.exclusive_lock->is_lock_owner());
 
   // immediately add a reference to the new snapshot
+  utime_t snap_time = ceph_clock_now();
   image_ctx.add_snap(m_snap_name, m_snap_namespace, m_snap_id, m_size, m_parent_info,
-                     RBD_PROTECTION_STATUS_UNPROTECTED, 0);
+                     RBD_PROTECTION_STATUS_UNPROTECTED, 0, snap_time);
```

Comment:
Why can't the cls_rbd method inject the timestamp -- no need to pass it from librbd